Update state version and workspace API docs (#379)
hashicorp/terraform-docs-common: website/docs/cloud-docs/api-docs/workspaces.mdx

## Unlock a workspace

> This endpoint unlocks a workspace. If the workspace is using Terraform >=v1.5.0, unlocking a workspace will set the current state version if any.

Reviewer: Is this behavior version dependent??

> This endpoint unlocks a workspace. Unlocking a workspace will set the current state version if there are any provisional state versions available.

Reviewer 2: will it set the current state version to the latest snapshot?

we might need to define "provisional state" in the docs (both here, and perhaps in workspace docs) so it's clear

> This endpoint unlocks a workspace. Unlocking a workspace will set the current state version if there are any provisional state versions available.

Reviewer 2: ```suggestion
This endpoint unlocks a workspace. Unlocking a workspace will set the current state version to the latest available finalized state version. If a finalized version is not available, unlock will fail.
```

based on my understanding of the comment below

## Force Unlock a workspace

> This endpoint force unlocks a workspace. Only users with admin access are authorized to force unlock a workspace. If the workspace is using Terraform >=v1.5.0, force unlocking the workspace will attempt to find the latest finalized state version to set as the current if any. It is possible that the latest finalized is not the most recently created state version.

Reviewer: Same as above: Is this behavior version dependent?

> This endpoint force unlocks a workspace. Only users with admin access are authorized to force unlock a workspace. Force unlocking the workspace will attempt to find the latest finalized state version to set as the current if there are any provisional state versions available. It is possible that the latest finalized state version is not the most recently created state version.

Reviewer 2: does the second sentence mean that it will look for a provisional state snapshot and use that instead of a full state file? if so, will suggest a slight rephrase so it's clear.

also, what are the implications of this for future runs?

Author: > does the second sentence mean that it will look for a provisional state snapshot and use that instead of a full state file? if so, will suggest a slight rephrase so it's clear.

State versions are the same as before but now that we've decoupled creating the state version record (i.e the metadata about the statefile) and uploading the actual state file. A state version that is `finalized` has both the record created and state file contents uploaded. A version where the record is created but the contents have not been uploaded is a `pending` state version. The attribute `status` tracks this. 

So during a regular unlock, TFC will ensure that the most recently created state version is finalized before setting it as the current state version for a workspace. If not, it will prevent an unlock. However something can go awry with the statefile upload which leaves the user no choice but to force unlock. In these cases, we will still attempt to find the most recent finalized version, this time without a hard stop at the latest state version. 

> also, what are the implications of this for future runs?
None. All this logic occurs before the next pending run is started. As long as we have a current state version set after an unlock, we're good.

> This endpoint force unlocks a workspace. Only users with admin access are authorized to force unlock a workspace. Force unlocking the workspace will attempt to find the latest finalized state version to set as the current if there are any provisional state versions available. It is possible that the latest finalized state version is not the most recently created state version.

Reviewer 2: ```suggestion
This endpoint force unlocks a workspace. Only users with admin access are authorized to force unlock a workspace. 

For a force unlock, Terraform Cloud checks for the latest finalized state version to set as the current state version. If the workspace has a provisional state version that is not yet finalized, the latest finalized state version may not be the most recently created state version.
```

This endpoint force unlocks a workspace. Only users with admin access are authorized to force unlock a workspace.

> For a force unlock, Terraform Cloud will attempt to find the latest provisional state version that has been finalized and set it as the current state version for a workspace. It is possible that an earlier provisional state version is selected as the current version for a workspace. This only occurs if the latest state version has not been finalized. Terraform Cloud will not prevent a force unlock if there are no available provisional state versions that have been finalized.

Reviewer 3: Suggestion for brevity (please correct me if this isn't technical correct though!):
```suggestion
For a force unlock, Terraform Cloud attempts to find the latest finalized provisional state version and sets it as the current state version for a workspace. Terraform Cloud may select an earlier provisional state as the current version for a workspace. This only occurs if Terraform Cloud has not yet finalized the latest state version. Terraform Cloud will not prevent a force unlock if there are no available finalized provisional state versions.
```
